bolt/lib/Target/AArch64/AArch64MCPlusBuilder.cpp

```diff
@@ -1140,7 +1300,248 @@ class AArch64MCPlusBuilder : public MCPlusBuilder {
     Inst.addOperand(MCOperand::createImm(0));
   }
 
-  bool mayStore(const MCInst &Inst) const override { return false; }
+  bool isPrefetch(const MCInst &Inst) const {
+    const unsigned opcode = Inst.getOpcode();
+    bool isNeedle = false;
+    switch (opcode) {
+    case AArch64::PRFMl:
+    case AArch64::PRFMui:
+    case AArch64::PRFUMi:
+    case AArch64::PRFMroW:
+    case AArch64::PRFMroX:
+      isNeedle = true;
+      break;
+
+    default:
+      break;
+    }
+
+    return isNeedle;
+  };
+
```

Comment:
And for stores:
  bool analyzeStore(const MCInst &Inst, MCPhysReg &From1, int64_t &Imm,
                    uint64_t &Count) const {
    // TODO add support of reg ldr, e.g. STRXroX, STRXroW
    From1 = AArch64::NoRegister;
    Imm = 0;
    Count = 1;
    switch(Inst.getOpcode()) {
    default:
      return false;
    case AArch64::STRXui:
    case AArch64::STRWui:
    case AArch64::STRHHui:
    case AArch64::STRBBui: {
      if (!Inst.getOperand(2).isImm())
        return false;
      From1 = Inst.getOperand(1).getReg();
      Imm = Inst.getOperand(2).getImm();
      break;
    }
    case AArch64::STURXi:
    case AArch64::STURWi:
    case AArch64::STURHHi:
    case AArch64::STURBBi: {
      if (!Inst.getOperand(2).isImm())
        return false;
      From1 = Inst.getOperand(1).getReg();
      Imm = Inst.getOperand(2).getImm();
      break;
    }
    case AArch64::STRXpre:
    case AArch64::STRWpre:
    case AArch64::STRHHpre:
    case AArch64::STRBBpre: {
      if (!Inst.getOperand(3).isImm())
        return false;
      From1 = Inst.getOperand(2).getReg();
      Imm = Inst.getOperand(3).getImm();
      break;
    }
    case AArch64::STRXpost:
    case AArch64::STRWpost:
    case AArch64::STRHHpost:
    case AArch64::STRBBpost: {
      From1 = Inst.getOperand(2).getReg();
      break;
    }
    case AArch64::STPXi:
    case AArch64::STPWi: {
      if (!Inst.getOperand(3).isImm())
        return false;
      From1 = Inst.getOperand(2).getReg();
      Imm = Inst.getOperand(3).getImm();
      Count = 2;
      break;
    }
    case AArch64::STPXpre:
    case AArch64::STPWpre: {
      if (!Inst.getOperand(4).isImm())
        return false;
      From1 = Inst.getOperand(3).getReg();
      Imm = Inst.getOperand(4).getImm();
      Count = 2;
      break;
    }
    case AArch64::STPXpost:
    case AArch64::STPWpost: {
      From1 = Inst.getOperand(3).getReg();
      Count = 2;
      break;
    }
    }

    switch(Inst.getOpcode()) {
      case AArch64::LDRXui:
      case AArch64::LDPXi:
      case AArch64::LDPXpre:
        Imm <<= 3;
        break;
      case AArch64::LDRWui:
      case AArch64::LDPWi:
      case AArch64::LDPSWi:
      case AArch64::LDPWpre:
      case AArch64::LDPSWpre:
        Imm <<= 2;
        break;
    }

    return true;
  }